src/main/java/exceptions/DataException.java

```diff
@@ -0,0 +1,26 @@
+package exceptions;
+
+/**
+ * Exception representing an error in data validation.
+ */
+public class DataException extends IPException {
+
+    private final String command;
+    private final String reason;
+
+    /**
+     * Creates a new exception.
+     * @param field the fields with invalid data
+     * @param reason the reason why the data is invalid
+     */
+    public DataException(String field, String reason) {
+        super(reason);
+        this.command = field;
```

Comment:
If possible, I would prefer to keep the `this` to make it clear that I am referring to the instance variable. Does the coding standard have anything that says otherwise?

In any case, I have renamed the variable to `field` as well. The original name `command` was a mistake from copying `CommandException`.